src/consts.rs

```diff
@@ -5,3 +5,117 @@ pub const KERNEL_OFFSET: usize = 0xFFFF_0000_0000_0000;
 
 pub const RAW_IO_BASE: usize = 0x3F00_0000;
 pub const IO_BASE: usize = KERNEL_OFFSET + RAW_IO_BASE;
+
+
+pub const CLOCK_FREQ	: usize	= 500000000;		// PLLD
+pub const CLOCK_DIVIDER	: usize	= 2;
+
+pub const ARM_IO_BASE	: usize = KERNEL_OFFSET + 0x3F000000;
+pub const ARM_DMA_BASE : usize = ARM_IO_BASE + 0x7000;
+// pub const ARM_DMA_CHAN_BASE : usize = KERNEL_OFFSET + 0x7E007000;  
+pub const GPU_IO_BASE	: usize = KERNEL_OFFSET + 0x7E000000;
```

Comment:
I think the address starts from `0x7E000000` is same as `0x3F000000`, so could the constant `GPU_IO_BASE ` be removed?